actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -709,3 +735,19 @@ func findByChannelLookup(ctx context.Context, storage exec.Storage, byPayer exec
 
 	return actor.LoadTypedLookup(ctx, storage, byChannelCID, &PaymentChannel{})
 }
+
+func checkCondition(vmctx exec.VMContext, condition *types.Predicate, redeemerSuppliedParams []interface{}) (uint8, error) {
+	if condition == nil {
+		return 0, nil
+	}
+	params := append(condition.Params[:0:0], condition.Params...)
+	params = append(params, redeemerSuppliedParams...)
+	_, _, err := vmctx.Send(condition.To, condition.Method, types.NewZeroAttoFIL(), params)
+	if err != nil {
+		if errors.IsFault(err) {
+			return errors.CodeError(err), err
+		}
+		return ErrConditionInvalid, errors.RevertErrorWrap(err, "failed to validate voucher condition")
```

Comment:
Dumb question: is it ok for this error to absorb out of gas errors?  I guess the error is wrapped so the user will see that the voucher isn't necessarily invalid, its just they ran out of gas?

Reply:
If the condition runs out of gas, we should see something like "failed to validate voucher condition: Insufficient gas". I think this is a pretty accurate description of the problem.

There are actually a lot of ways this message can fail that don't involve the condition being invalid. Hopefully we've covered most of them in the tests. The burden is on the target to ensure that payment vouchers have valid conditions and they call the methods with enough gas. We'll have two features to support this:

1. We'll add a method to payment broker to test a condition off chain with a query method.
2. We'll have storage miners automatically check all payment vouchers have valid conditions once they have all the data (correct miner address, correct method, and correct CommP).